admin-dev/themes/new-theme/js/components/addons-connector.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+const $ = window.$;
+
+export default class AddonsConnector {
+  constructor(
+    addonsConnectFormSelector,
+    addonsLoginButtonSelector
+  ) {
+    this.addonsConnectFormSelector = addonsConnectFormSelector;
+    this.$addonsLoginButton = $(addonsLoginButtonSelector);
+
+    this.initEvents();
+  }
+
+  /**
+   * Initialize events related to connection to addons.
+   */
+  initEvents() {
+    const t = this;
+
+    $('body').on('submit', this.addonsConnectFormSelector, function (event) {
+      event.preventDefault();
+      event.stopPropagation();
+
+      t._connect($(this).attr('action'), $(this).serialize());
+    });
+  }
+
+  /**
+   * Do a POST request to connect to addons.
+   *
+   * @private
+   */
+  _connect(addonsConnectUrl, formData) {
+    $.ajax({
+      method: 'POST',
+      url: addonsConnectUrl,
+      dataType: 'json',
+      data: formData,
+      beforeSend: () => {
+        this.$addonsLoginButton.show();
+        $('button.btn[type="submit"]', this.addonsConnectFormSelector).hide();
+      }
+    }).done((response) => {
```

Comment:
if we don't handle `fail()` it will probably just fail silently with an error in console. `fail()` is not likely to happen though, only if the addons server would fail to respond, should we take care of such edge case?

If user enters invalid credentials, he will get a nice error message already.

Reply:
I've just checked - this ajax request is actually sent to prestashop's backend, which then sends another request to addons API and handles any errors that occurred, so if addons server fails we will get an error message already

Reply:
It does not change :) we must still handle the usecase "the shop BO takes too much time to return a response or crashes and returns a HTTP 500" with a nice error message. Allowing the script to fail with an error in console is going to make all JS on the page errored, and might freeze the page.

Reply:
fixed